section_faiss_vector_index.go

```diff
@@ -564,32 +564,41 @@ func (vo *vectorIndexOpaque) process(field index.VectorField, fieldID uint16, do
 	dim := field.Dims()
 	metric := field.Similarity()
 
-	if vec != nil {
+	// Embedder is supposed to make sure len(vec) is a multiple of dim.
+	// Not double checking it here to avoid the overhead.
+	numSubVecs := len(vec) / int(dim)
+	for i := 0; i < numSubVecs; i++ {
```

Comment:
now i don't have any benchmark numbers of something to suport this doubt, however in a purely non-nested situation where the loop is iterated only once, could there be a perf impact due to doing the following code as part of for loop iteration rather than just doing it as part of a conditional check of `if len(vec) == int(dim)`. 

i dont know too much about the golang internals, but yeah just a doubt i had when i glanced through the code, so just a small food for thought ig

Reply:
Thanks. 
Will continue this discussion in the next PR ( targeting some optimizations)